### PR TITLE
[minor_change] support for dhcpLabels

### LINKED ISSUE #158: [mso_schema_template_bd] Allow multiple dhcp_policy blocks to support NDO
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

In recent version of NDO, multiple DHCP policies can be associated with a BD. We need to update the mso_schema_template_bd resource to support multiple blocks.

We need to make sure it does not brake updates from older version of the provider.

### New or Affected Resource(s) + MSO Class(es):

* mso_schema_template_bd

### MSO version

* V 3.7

### APIC version and APIC Platform for Site level Resources

* on-prem

### Potential Terraform Configuration

Make it a block and not just a map.

### PR DESCRIPTION
**Pull request by akinross**
fixes #158

### CODE REVIEW

File: examples/schema_template_bd/main.tf [hunk 1]
```diff
@@ -49,12 +51,36 @@ resource "mso_schema_template_bd" "bd" {
   multi_destination_flooding = "flood_in_encap"
   ipv6_unknown_multicast_flooding = "optimized_flooding"
   unknown_multicast_flooding = "optimized_flooding"
-  dhcp_policy = {
+  dhcp_policies {
```
Comment (shrsr) [inline examples/schema_template_bd/main.tf:54]: Is is possible to add another block of dhcp policy here?

File: mso/resource_mso_schema_template_bd.go [hunk 2]
```diff
@@ -618,110 +767,59 @@ func resourceMSOTemplateBDUpdate(d *schema.ResourceData, m interface{}) error {
 		dhcpPolMap = nil
 	}
 
+	dhcpPolList := make([]interface{}, 0)
```
Comment (shrsr) [inline mso/resource_mso_schema_template_bd.go:770]: Does update work as expected when configuring multiple blocks of dhcp policies? -
Example -
Add blocks, Remove blocks one at a time etc.

File: website/docs/r/schema_template_bd.html.markdown [hunk 3]
```diff
@@ -45,13 +45,20 @@ resource "mso_schema_template_bd" "bridge_domain" {
 * `ipv6_unknown_multicast_flooding` - (Optional) IPv6 Unknown Multicast Flooding behavior. Allowed values are `flood` and `optimized_flooding`. Default to `flood`.
 * `multi_destination_flooding` - (Optional) Multi-destination flooding behavior. Allowed values are `flood_in_bd`, `drop` and `flood_in_encap`. Default to `flood_in_bd`.
 * `unknown_multicast_flooding` - (Optional) Unknown Multicast Flooding behavior. Allowed values are `flood` and `optimized_flooding`. Default to `flood`.
+* `dhcp_policies` - (Optional) Block to provide dhcp_policy configurations. 
+* `dhcp_policies.name` - (Optional) Dhcp_policy name. Required if you specify the dhcp_policy.
+* `dhcp_policies.version` - (Optional) Version of dhcp_policy. Required if you specify the dhcp_policy.
+* `dhcp_policies.dhcp_option_policy_name` - (Optional) Name of dhcp_option_policy. 
+* `dhcp_policies.dhcp_option_policy_version` - (Optional) Version of dhcp_option_policy.
```
Comment (sajagana) [inline website/docs/r/schema_template_bd.html.markdown:52]: Please change the documentation format.
For your reference: https://github.com/CiscoDevNet/terraform-provider-aci/pull/857/files#diff-0505816fc7d3615c5e4a081b8c2cf3e731a4b18fdb4d76f12bcf6e67ac7d3de1R107-R111

File: website/docs/d/schema_template_bd.html.markdown [hunk 4]
```diff
@@ -40,7 +40,15 @@ data "mso_schema_template_bd" "bridge_domain" {
 * `optimize_wan_bandwidth` - (Optional) Boolean flag to enable or disable the wan bandwidth optimization.
 * `layer2_stretch` - (Optional) Boolean flag to enable or disable the layer-2 stretch.
 * `layer3_multicast` - (Optional) Boolean flag to enable or disable layer 3 multicast traffic.
-* `dhcp_policy` - (Optional) Map to provide dhcp_policy configurations. 
+* `dhcp_policies` - (Optional) Block to provide dhcp_policy configurations. 
+* `dhcp_policies.name` - (Optional) Dhcp_policy name. Required if you specify the dhcp_policy.
+* `dhcp_policies.version` - (Optional) Version of dhcp_policy. Required if you specify the dhcp_policy.
+* `dhcp_policies.dhcp_option_policy_name` - (Optional) Name of dhcp_option_policy. 
+* `dhcp_policies.dhcp_option_policy_version` - (Optional) Version of dhcp_option_policy.
```
Comment (sajagana) [inline website/docs/d/schema_template_bd.html.markdown:47]: Please change the documentation format.
For your reference: https://github.com/CiscoDevNet/terraform-provider-aci/pull/857/files#diff-0505816fc7d3615c5e4a081b8c2cf3e731a4b18fdb4d76f12bcf6e67ac7d3de1R107-R111

File: examples/schema_template_bd/main.tf (hunk 1 above)
Comment (sajagana) [reply examples/schema_template_bd/main.tf:54]: Also, update one sample!

File: website/docs/r/schema_template_bd.html.markdown (hunk 3 above)
Comment (akinross) [reply website/docs/r/schema_template_bd.html.markdown:52]: ok

File: website/docs/d/schema_template_bd.html.markdown (hunk 4 above)
Comment (akinross) [reply website/docs/d/schema_template_bd.html.markdown:47]: ok

File: mso/datasource_mso_schema_template_bd.go [hunk 5]
```diff
@@ -118,109 +138,34 @@ func dataSourceMSOTemplateBD() *schema.Resource {
 					},
 				},
 			},
+			"dhcp_policies": &schema.Schema{
```
[the remark this answers is not in the document]
Comment (akinross) [reply mso/datasource_mso_schema_template_bd.go:141]: ok

File: mso/resource_mso_schema_template_bd.go [hunk 6]
```diff
@@ -163,7 +164,42 @@ func resourceMSOTemplateBD() *schema.Resource {
 					},
 				},
 			},
+			"dhcp_policies": &schema.Schema{
```
[the remark this answers is not in the document]
Comment (akinross) [reply mso/resource_mso_schema_template_bd.go:167]: ok

**Review by sajagana — APPROVED** [review]
LGTM!

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**